src/serverless-openapi-typescript.ts

```diff
@@ -2,241 +2,267 @@ import type Serverless from "serverless";
+                const generatedSchemas = this.serverless.service.custom.documentation.models
+                const paramTypeName = `${definitionPrefix}.Request.${upperFirst(documentationKey)}.${name}`;
+                const haveSchema = !isEmpty(generatedSchemas.find(schemaName => schemaName.name === paramTypeName));
+                return haveSchema ? {$ref: `#/components/schemas/${paramTypeName}`} : {type: 'string'}
+            }
+
+            const schema = getParamSchema();
+            const paramDocumentationFromSls = {
+                name,
+                required,
+                schema
+            };
+
+            if (!existingDocumentedParam) {
+                documentedParams.push(paramDocumentationFromSls);
+            } else {
+                Object.assign(paramDocumentationFromSls, existingDocumentedParam);
+                Object.assign(existingDocumentedParam, paramDocumentationFromSls);
+            }
+        });
+    }
+
+    setModels(httpEvent, functionName, querystrings = {}) {
+        const definitionPrefix = this.getDefinitionPrefix(functionName);
+        const method = httpEvent.method.toLowerCase();
+        switch (method) {
+            case 'delete':
+                set(httpEvent, 'documentation.methodResponses', [{statusCode: 204, responseModels: {}}]);
+                break;
+            case 'patch':
+            case 'put':
+            case 'post':
+                const requestModelName = `${definitionPrefix}.Request.Body`;
+                this.setModel(`${definitionPrefix}.Request.Body`);
+                set(httpEvent, 'documentation.requestModels', {'application/json': requestModelName});
+                set(httpEvent, 'documentation.requestBody', {description: ''});
+            // no-break;
+            case 'get':
+                const responseModelName = `${definitionPrefix}.Response`;
+                this.setModel(responseModelName);
+                this.setQueryStringsModel(querystrings, definitionPrefix);
+                set(httpEvent, 'documentation.methodResponses', [
+                    {
+                        statusCode: 200,
+                        responseBody: {description: ''},
+                        responseModels: {'application/json': responseModelName}
+                    }
+                ]);
         }
-      },
-      (objValue, srcValue) => {
-        if (isArray(objValue)) {
-          return objValue.concat(srcValue);
+    }
+
+    private setQueryStringsModel(querystrings: Record<string, boolean>, definitionPrefix: string) {
+        if (!isEmpty(querystrings)) {
+            keys(querystrings).forEach(entry => {
+                const queryParamsModelName = `${definitionPrefix}.Request.QueryParams.${entry}`;
+                try {
+                    this.setModel(queryParamsModelName);
+                } catch (err) {
+                    this.log(`Didn't find ${queryParamsModelName}`);
```

Comment:
```suggestion
                    this.log(`Didn't find ${queryParamsModelName}, skipping override.`);
```

Reply:
its skipping type generation
ill add that to the log a11e16d1